### PR TITLE
at-cc: Support HSC and DC-to-DC power module device

### PR DESCRIPTION
**Pull request by DelphineChiu**
Summary:
- Support HSC (ADM1272) device.
- Support DC-to-DC power module (Q50SN120A1) device.

Test Plan:
- Build code: Pass

### CODE REVIEW

File: common/dev/adm1272.c [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+	case PMBUS_READ_VOUT:
+		switch (vrange) {
+		case VRANGE_0V_TO_60V:
+			convert_coefficient_m = 6770;
+			break;
+		case VRANGE_0V_TO_100V:
+			convert_coefficient_m = 4062;
+			break;
+		default:
+			LOG_ERR("convert value func: vrange 0x%x is invlid", vrange);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:47]: Remove function name
```
LOG_ERR("Vrange 0x%x is invlid", vrange);
```

File: common/dev/adm1272.c [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	case PMBUS_READ_IOUT:
+		switch (irange) {
+		case IRANGE_0MV_TO_15MV:
+			convert_coefficient_m = 1326;
+			break;
+		case IRANGE_0MV_TO_30MV:
+			convert_coefficient_m = 663;
+			break;
+		default:
+			LOG_ERR("convert value func: irange 0x%x is invlid", irange);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:63]: Remove function name at the beginning of the log.

File: common/dev/adm1272.c [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+			case IRANGE_0MV_TO_15MV:
+				convert_coefficient_m = 3512;
+				convert_coefficient_R = 100;
+				break;
+			case IRANGE_0MV_TO_30MV:
+				convert_coefficient_m = 17561;
+				convert_coefficient_R = 1000;
+				break;
+			default:
+				LOG_ERR("convert value func: irange 0x%x is invlid", irange);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:89]: Remove function name at the beginning of the log.

File: common/dev/adm1272.c [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+			case IRANGE_0MV_TO_15MV:
+				convert_coefficient_m = 21071;
+				convert_coefficient_R = 1000;
+				break;
+			case IRANGE_0MV_TO_30MV:
+				convert_coefficient_m = 10535;
+				convert_coefficient_R = 1000;
+				break;
+			default:
+				LOG_ERR("convert value func: irange 0x%x is invlid", irange);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:104]: Remove function name at the beginning of the log.

File: common/dev/adm1272.c [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+				return -1;
+			}
+			break;
+		default:
+			LOG_ERR("convert value func: vrange 0x%x is invlid", vrange);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:109]: Remove function name at the beginning of the log.

File: common/dev/adm1272.c [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+			return -1;
+		}
+		convert_coefficient_b = 0;
+		break;
+
+	default:
+		LOG_ERR("convert func: offset 0x%x is invalid", offset);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:116]: Remove function name at the beginning of the log.

File: common/dev/adm1272.c [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+		return -1;
+	}
+
+	if (convert_coefficient_m == 0) {
+		LOG_ERR("convert func: convert coefficient m is invalid, vrange: 0x%x, irange: 0x%x",
```
Comment (GoldenBug) [inline common/dev/adm1272.c:121]: Remove function name at the beginning of the log.

File: common/dev/adm1272.c [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+	I2C_MSG msg = { 0 };
+	msg.bus = cfg.port;
+	msg.target_addr = cfg.target_addr;
+	msg.tx_len = 1;
+	msg.rx_len = 2;
+	msg.data[0] = PMBUS_READ_VOUT;
+
+	ret = i2c_master_read(&msg, retry);
+	if (ret != 0) {
+		LOG_ERR("read pout func: read Vout fail, ret: %d", ret);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:157]: Remove function name at the beginning of the log.

File: common/dev/adm1272.c [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+	memset(&msg, 0, sizeof(msg));
+	msg.bus = cfg.port;
+	msg.target_addr = cfg.target_addr;
+	msg.tx_len = 1;
+	msg.rx_len = 2;
+	msg.data[0] = PMBUS_READ_IOUT;
+
+	ret = i2c_master_read(&msg, retry);
+	if (ret != 0) {
+		LOG_ERR("read pout func: read Iout fail, ret: %d", ret);
```
Comment (GoldenBug) [inline common/dev/adm1272.c:178]: Remove function name at the beginning of the log.

File: common/dev/q50sn120a1.c [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+		val = val * Q50SN120A1_TEMPERATURE_SCALE;
+		break;
+	case PMBUS_READ_POUT:
+		ret = q50sn120a1_read_pout(sensor_num, &val);
+		if (ret != 0) {
+			return SENSOR_FAIL_TO_ACCESS;
+		}
+		break;
+	default:
+		LOG_ERR("read func: not support offset: 0x%x", offset);
```
Comment (GoldenBug) [inline common/dev/q50sn120a1.c:121]: Remove function name at the beginning of the log.

File: common/dev/q50sn120a1.c [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+	sval->integer = (int)val & 0xFFFF;
+	sval->fraction = (val - sval->integer) * 1000;
+	return SENSOR_READ_SUCCESS;
+}
+
+uint8_t q50sn120a1_init(uint8_t sensor_num)
+{
+	if (sensor_num > SENSOR_NUM_MAX) {
+		LOG_ERR("init func: input sensor number is invalid");
```
Comment (GoldenBug) [inline common/dev/q50sn120a1.c:134]: Remove function name at the beginning of the log.

**Review by GoldenBug — CHANGES_REQUESTED** [review]

**Comment by LoraLin1** [comment]
Hi @GoldenBug ,
Thanks for your comments.
Based on your suggestions to modify the code, rebase main and force update this branch.
Please help to review it again, thank you.

**Pull request closed without merging**